### ISSUE TITLE
Plugin 1.2.1.1, ES 1.2.1: ElasticsearchIllegalStateException: client is closed

### ISSUE THREAD

**Issue by jowy** [jprante/elasticsearch-jdbc#264]
TLDR: problem found in **Update 3** below

Initial run works fine. Consecutive scheduled and manual runs (re-put/delete) give this error and nothing is ran. Deleting the river and associated index will cause it to run once again, then same error.

```
curl -XPUT 'elasticsearch001.my.company.lan:9200/_river/publishers/_meta' -d '{
    "type" : "jdbc",
    "jdbc" : {
        "schedule" : "*/10 * * * * ?",
        "url" : "jdbc:mysql://hamysql.my.company.lan:13306/gibson",
        "user" : "root",
        "password" : "hunter2",
        "sql" : "SELECT `id` AS `_id`, UNIX_TIMESTAMP(`updated_at`) as `_version`, ... "
    }
}'
```

```
[2014-06-19 00:39:32,328][INFO ][cluster.metadata         ] [elasticsearch001] [_river] creating index, cause [auto(index api)], shards [5]/[0], mappings []
[2014-06-19 00:39:32,408][INFO ][cluster.metadata         ] [elasticsearch001] [_river] update_mapping [my_river] (dynamic)
[2014-06-19 00:39:32,415][INFO ][Feeder                   ] scheduled feeder with cron expressions [*/10 * * * * ?]
[2014-06-19 00:39:32,421][INFO ][cluster.metadata         ] [elasticsearch001] [_river] update_mapping [my_river] (dynamic)

[2014-06-19 00:39:40,001][INFO ][JDBCFeeder               ] river default index/type jdbc/jdbc (timewindowed=false)
[2014-06-19 00:39:40,002][INFO ][JDBCFeeder               ] creating index jdbc and enabling bulk mode
[2014-06-19 00:39:40,002][INFO ][NodeClient               ] creating index jdbc with settings = {}, mappings = null
[2014-06-19 00:39:40,024][INFO ][cluster.metadata         ] [elasticsearch001] [jdbc] creating index, cause [api], shards [5]/[0], mappings []
[2014-06-19 00:39:40,092][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][0] updating refresh_interval from [1s] to [-1]
[2014-06-19 00:39:40,092][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][3] updating refresh_interval from [1s] to [-1]
[2014-06-19 00:39:40,092][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][1] updating refresh_interval from [1s] to [-1]
[2014-06-19 00:39:40,092][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][2] updating refresh_interval from [1s] to [-1]
[2014-06-19 00:39:40,092][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][4] updating refresh_interval from [1s] to [-1]
[2014-06-19 00:39:40,111][INFO ][cluster.metadata         ] [elasticsearch001] [_river] update_mapping [my_river] (dynamic)
[2014-06-19 00:39:40,197][INFO ][cluster.metadata         ] [elasticsearch001] [jdbc] update_mapping [jdbc] (dynamic)
[2014-06-19 00:39:40,278][INFO ][NodeClient               ] flushing bulk processor
[2014-06-19 00:39:40,351][INFO ][JDBCFeeder               ] stopping bulk mode for indices [jdbc]...
[2014-06-19 00:39:40,353][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][0] updating refresh_interval from [-1] to [1s]
[2014-06-19 00:39:40,353][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][3] updating refresh_interval from [-1] to [1s]
[2014-06-19 00:39:40,353][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][1] updating refresh_interval from [-1] to [1s]
[2014-06-19 00:39:40,354][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][2] updating refresh_interval from [-1] to [1s]
[2014-06-19 00:39:40,354][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][4] updating refresh_interval from [-1] to [1s]
[2014-06-19 00:39:40,355][INFO ][Feeder                   ] close (no-op)
[2014-06-19 00:39:40,355][INFO ][JDBCFeeder               ] reading connection closed
[2014-06-19 00:39:40,355][INFO ][JDBCFeeder               ] writing connection closed
[2014-06-19 00:39:40,355][INFO ][JDBCFeeder               ] river mouth closing
[2014-06-19 00:39:40,355][INFO ][JDBCFeeder               ] river mouth closed
[2014-06-19 00:39:40,365][INFO ][JDBCFeeder               ] river state saved
[2014-06-19 00:39:40,365][INFO ][Feeder                   ] shutting down executor
[2014-06-19 00:39:40,365][INFO ][NodeClient               ] flushing bulk processor

[2014-06-19 00:39:50,002][INFO ][JDBCFeeder               ] river default index/type jdbc/jdbc (timewindowed=false)
[2014-06-19 00:39:50,004][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][0] updating refresh_interval from [1s] to [-1]
[2014-06-19 00:39:50,004][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][3] updating refresh_interval from [1s] to [-1]
[2014-06-19 00:39:50,004][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][1] updating refresh_interval from [1s] to [-1]
[2014-06-19 00:39:50,004][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][2] updating refresh_interval from [1s] to [-1]
[2014-06-19 00:39:50,005][INFO ][index.shard.service      ] [elasticsearch001] [jdbc][4] updating refresh_interval from [1s] to [-1]
[2014-06-19 00:39:50,087][ERROR][NodeClient               ] bulk [4] failed
[2014-06-19 00:39:50,091][ERROR][Feeder                   ] error while getting next input: org.elasticsearch.ElasticsearchIllegalStateException: client is closed
java.io.IOException: org.elasticsearch.ElasticsearchIllegalStateException: client is closed
    at org.xbib.elasticsearch.river.jdbc.strategy.simple.SimpleRiverSource.fetch(SimpleRiverSource.java:248)
    at org.xbib.elasticsearch.plugin.feeder.jdbc.JDBCFeeder.executeTask(JDBCFeeder.java:159)
    at org.xbib.elasticsearch.plugin.feeder.AbstractFeeder.newRequest(AbstractFeeder.java:361)
    at org.xbib.elasticsearch.plugin.feeder.AbstractFeeder.newRequest(AbstractFeeder.java:52)
    at org.xbib.pipeline.AbstractPipeline.call(AbstractPipeline.java:86)
    at org.xbib.pipeline.AbstractPipeline.call(AbstractPipeline.java:17)
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:744)
Caused by: org.elasticsearch.ElasticsearchIllegalStateException: client is closed
    at org.xbib.elasticsearch.support.client.node.NodeClient.index(NodeClient.java:198)
    at org.xbib.elasticsearch.support.client.node.NodeClient.index(NodeClient.java:38)
    at org.xbib.elasticsearch.river.jdbc.strategy.simple.SimpleRiverMouth.index(SimpleRiverMouth.java:140)
    at org.xbib.elasticsearch.plugin.jdbc.RiverMouthKeyValueStreamListener.end(RiverMouthKeyValueStreamListener.java:38)
    at org.xbib.elasticsearch.plugin.jdbc.RiverMouthKeyValueStreamListener.end(RiverMouthKeyValueStreamListener.java:11)
    at org.xbib.elasticsearch.plugin.jdbc.PlainKeyValueStreamListener.values(PlainKeyValueStreamListener.java:124)
    at org.xbib.elasticsearch.river.jdbc.strategy.simple.SimpleRiverSource.processRow(SimpleRiverSource.java:644)
    at org.xbib.elasticsearch.river.jdbc.strategy.simple.SimpleRiverSource.nextRow(SimpleRiverSource.java:601)
    at org.xbib.elasticsearch.river.jdbc.strategy.simple.SimpleRiverSource.merge(SimpleRiverSource.java:380)
    at org.xbib.elasticsearch.river.jdbc.strategy.simple.SimpleRiverSource.execute(SimpleRiverSource.java:281)
    at org.xbib.elasticsearch.river.jdbc.strategy.simple.SimpleRiverSource.fetch(SimpleRiverSource.java:243)
    ... 9 more
[2014-06-19 00:39:50,091][INFO ][Feeder                   ] close (no-op)
[2014-06-19 00:39:50,091][INFO ][JDBCFeeder               ] reading connection closed
[2014-06-19 00:39:50,091][INFO ][JDBCFeeder               ] writing connection closed
[2014-06-19 00:39:50,091][INFO ][JDBCFeeder               ] river mouth closing
[2014-06-19 00:39:50,091][INFO ][JDBCFeeder               ] river mouth closed
[2014-06-19 00:39:50,103][INFO ][JDBCFeeder               ] river state saved
[2014-06-19 00:39:50,104][INFO ][Feeder                   ] shutting down executor
```
## Update 1

Got some debug info:

This is at the end of the **first run** (successful) of 2642 total objects:

```
[2014-06-19 04:59:42,661][DEBUG][NodeClient               ] after bulk [14] [succeeded=2642] [failed=0] [67ms]
[2014-06-19 04:59:42,661][DEBUG][JDBCFeeder               ] flushed
[2014-06-19 04:59:42,661][INFO ][JDBCFeeder               ] stopping bulk mode for indices [jdbc]...
...
[2014-06-19 04:59:42,666][DEBUG][cluster.service          ] [elasticsearch001] processing [update-settings]: done applying updated cluster_state (version: 16)
```

**Second run:**

```
[2014-06-19 04:59:50,010][DEBUG][cluster.service          ] [elasticsearch001] cluster state updated, version [17], source [update-settings]
...
[2014-06-19 04:59:50,099][DEBUG][NodeClient               ] before bulk [15] of 200 items, 30815 bytes, 0 outstanding bulk requests
[2014-06-19 04:59:50,126][ERROR][NodeClient               ] bulk [15] failed
[2014-06-19 04:59:50,126][DEBUG][NodeClient               ] after bulk [15] [succeeded=2642] [failed=200] [26ms]
[2014-06-19 04:59:50,149][ERROR][Feeder                   ] error while getting next input: org.elasticsearch.ElasticsearchIllegalStateException: client is closed
```

The "[2014-06-19 04:59:50,126][DEBUG]" line looks suspicious. On the first bulk of the second run, we're already at 2642 succeeded, with 200 failed? Weird. **Note: This is using "maxbulkactions":"200"**

Client is being closed here: https://github.com/jprante/elasticsearch-support/blob/master/src/main/java/org/xbib/elasticsearch/support/client/node/NodeClient.java#L134

---

Second run with maxbulkactions:10000:

```
[2014-06-19 05:21:30,369][DEBUG][org.xbib.elasticsearch.river.jdbc.strategy.simple.SimpleRiverSource] merged 2642 rows
[2014-06-19 05:21:30,370][DEBUG][JDBCFeeder               ] fetched, flushing
[2014-06-19 05:21:30,371][INFO ][NodeClient               ] flushing bulk processor
[2014-06-19 05:21:30,371][DEBUG][NodeClient               ] before bulk [2] of 2642 items, 405042 bytes, 0 outstanding bulk requests
[2014-06-19 05:21:30,673][ERROR][NodeClient               ] bulk [2] failed
[2014-06-19 05:21:30,675][DEBUG][NodeClient               ] after bulk [2] [succeeded=2642] [failed=2642] [301ms]
```
## Update 2

This is also happening on a fresh install of Ubuntu 12.04 LTS w/ Plugin 1.2.1.1 and ES 1.2.1

```
java version "1.7.0_55"
OpenJDK Runtime Environment (IcedTea 2.4.7) (7u55-2.4.7-1ubuntu1~0.12.04.2)
OpenJDK 64-Bit Server VM (build 24.51-b03, mixed mode)
```
## Update 3

I have discovered that it fails if I specify MySQL field aliases both "_version" and "_id." It will only work with one or the other. Is this not supported? Sorry, now that I have found this issue, I do not have time to go through the Java code. Cheers!
